### PR TITLE
Add parser to sanitize-html's IOptions interface

### PR DESCRIPTION
**Pull request by WillGibson**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/punkave/sanitize-html/issues/248
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/sanitize-html/index.d.ts
Comment (BehindTheMath) [reply types/sanitize-html/index.d.ts]: I don't think I've ever found documentation on how to write definitions that depends on those of another package.

I think you should try importing it in the beginning of the code:
```
import {Options} from "htmlparser2";
```
and then define `parser` as type `Options`:

```
parser: Options
```

Then write a test that uses it. If it passes, it should be good.

Comment (boutell) [reply types/sanitize-html/index.d.ts]: Hmm I see you are still discussing with someone who knows their typescript, so I'll hold off on merge. Let me know.

Comment (WillGibson) [reply types/sanitize-html/index.d.ts]: I'll test the proposal in the morning.

Comment (WillGibson) [reply types/sanitize-html/index.d.ts]: @BehindTheMath Your suggestion works, tested locally, build passed etc.

**Review by biermeester — APPROVED** [review]
Local test worked

**Review by boutell — APPROVED** [review]

**Comment by WillGibson** [comment]
@rogierschouten @afshin @BehindTheMath @biermeester This has two approvals now, does one of you have the right permissions to merge/publish to NPM?

**Comment by BehindTheMath** [comment]
A maintainer of the repo has to do it. Give them a few days.

**Comment by WillGibson** [comment]
Thanks for the merge @minestarks

**Pull request merged**
